### PR TITLE
Ishfaq/yup validation

### PR DESCRIPTION
**Pull request by IshfaqAhmedAwan**

### CODE REVIEW

File: src/StyledComponents/HomeStyle.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import Message from 'assets/images/MessageLogo.png';
+
+export const MessageIcon = styled.img`
+  margin-top: 44px;
+  margin-bottom: 0px;
+  margin-left: 10px;
+  width: 71px;
+  height: 53.99px;
+`;
+MessageIcon.defaultProps = { src: Message };
```
Comment (alitjanjua) [inline src/StyledComponents/HomeStyle.ts:11]: that's something new for me 👍

File: src/components/UserRegister.tsx [hunk 2]
```diff
@@ -135,115 +68,137 @@ const UserForm = (props: any) => {
...
+                    render={({ field, formState }) => (
+                      <StyledTextField
+                        variant={'outlined'}
+                        width='170px'
+                        height='58px'
+                        id='standard-basic'
+                        label='First Name'
+                        {...field}
+                        error={!!formState.errors?.firstName}
+                        helperText={!!formState.errors?.firstName ? 'Required or invalid' : ''}
```
Comment (alitjanjua) [inline src/components/UserRegister.tsx]: use error msg of yup schema!

File: src/components/UserRegister.tsx [hunk 3]
```diff
@@ -135,115 +68,137 @@ const UserForm = (props: any) => {
...
+                        height='58px'
                         id='standard-basic'
                         label='Phone Number'
-                        className={classes.phonenumber}
-                        {...register('phoneNumber', {
-                          required: true,
-                          pattern: phonePattern === 'AUS' ? /^[0-9\s]*$/i : /^(\([0-9]{3}\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$/i,
-                        })}
-                        helperText={!!errors?.phoneNumber ? 'invalid phone number' : ''}
+                        InputProps={{
```
Comment (alitjanjua) [inline src/components/UserRegister.tsx:136]: cool 💯

File: src/pages/User/validations/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import * as Yup from 'yup';
+
+export const signUpSchema = Yup.object().shape({
+  firstName: Yup.string().required(),
```
Comment (alitjanjua) [inline src/pages/User/validations/index.ts]: LIke you can declare error msg here like below
firstName: Yup.string().required('First Name is Required'),

**Review by alitjanjua — CHANGES_REQUESTED** [review]
You're improving and learning at very smart pace.
Good Luck.

File: src/StyledComponents/HomeStyle.ts (hunk 1 above)
Comment (IshfaqAhmedAwan) [reply src/StyledComponents/HomeStyle.ts:11]: oh Wow

File: src/components/UserRegister.tsx (hunk 3 above)
Comment (IshfaqAhmedAwan) [reply src/components/UserRegister.tsx:136]: Thank you Sir

**Review by alitjanjua — CHANGES_REQUESTED** [review]
everything looks good to me but just remove console statements.

File: src/components/UserRegister.tsx [hunk 5]
```diff
@@ -135,115 +65,144 @@ const UserForm = (props: any) => {
   };
```
Comment (alitjanjua) [inline src/components/UserRegister.tsx:69]: you don't have to use phonePattern state for any purpose
you can use it like this
`
if(Flags[index].code === '+1-United States'){
    setValue('country', 'US')
} else{
    ...
}
`

Comment (alitjanjua) [reply src/components/UserRegister.tsx:69]: also use enums

**Review by alitjanjua — CHANGES_REQUESTED** [review]

File: src/components/UserRegister.tsx [hunk 6]
```diff
@@ -23,227 +26,180 @@ const Flags = [
...
   }; // your form submit function which will invoke after successful validation
 
-  //Call back function for FlagSelect component
+  // Call back function for FlagSelect component
   const flagStateChange = (index: number) => {
     setisFlagDialog(false);
     setSelectedFlag(Flags[index].img);
-    Flags[index].code === '+1-United States' ? setPhonePattern('US') : setPhonePattern('AUS');
+    // Created a custum field in register validate according to country in yup schema
+    Flags[index].code === '+1-United States' ? setValue('country', countries.AMERICA) : setValue('country', countries.AUSTRALIA);
```
Comment (alitjanjua) [inline src/components/UserRegister.tsx]: ternary operator is used where you have to return something
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

Comment (IshfaqAhmedAwan) [reply src/components/UserRegister.tsx]: Got it done

**Review by alitjanjua — APPROVED** [review]
LGTM 👍

**Pull request merged**